GTMAppAuthSwift/Sources/GTMKeychain.swift

```diff
@@ -15,9 +15,348 @@
  */
 
 import Foundation
+import Security
 
-/// Utility for saving and loading data to the keychain.
-@objc
-public class GTMKeychain: NSObject {
+/// An internal utility for saving and loading data to the keychain.
+final class GTMKeychain {
+  private var keychainHelper: KeychainHelper
 
+  /// An initializer to create an instance of this keychain wrapper with a given
+  /// helper.
+  /// - parameter keychainHelper: An instance conforming to `KeychainHelper`.
+  /// - note: This initializer is here to help with testing.
+  init(keychainHelper: KeychainHelper? = nil) {
+    if let helper = keychainHelper {
+      self.keychainHelper = helper
+    } else {
+      self.keychainHelper = KeychainWrapper()
+    }
+  }
+
+  /// Saves the password `String` to the keychain with the given identifier.
+  /// - parameter password: The `String` password.
+  /// - parameter name: The name for the Keychain item.
+  /// - throws: An instance of `KeychainWrapper.Error`.
```

Comment:
I like the improved error handling here.  Regarding our (semi)strict translation approach here, what does this mean for the exported ObjC interface?  I believe an NSError parameter will be added, no?

Reply:
In a sense, I think this question is moot. `GTMKeychain` is `internal` now, and the plan is to translate/rewrite all of the types in GTMAppAuth to Swift. Furthermore, this method is not marked as `@objc`, and so it will not be exposed to Objective-C.

But yes, you're right. If this method were `@objc`, it would be exported to ObjC and called like so:

```
NSError *err;
[keychain saveWithPassword:@"foo" forName:@"name" error:&err];
```

Reply:
Oh yes, of course.  Sorry, briefly forgot that we are making `GTMKeychain` internal.  This interface will only be used in Swift going forward.